frontend/src/components/ManageAccess.vue

```diff
@@ -1,18 +1,146 @@
 
-var foo = 2;
+const apiService = new ApiService()
+
+interface UserRoleAssignment {
+  user_role_xref_id: number,
+  user: {
+    user_type_code: string
+    user_type_desc?:string
+    cognito_user_id: string
+    user_name: string
+  },
+  role: {
+    role_name: string
+    role_type_code: string
+    client_number?: {
+      client_name: string,
+      forest_client_number: string
+    }
+  }
+
+    // assignment_id: number;
+    // user_id: string;
+    // user_domain: string;
+    // role: string
+    // forest_client_number?: string;
+}
+
+const userRoleAssignments = ref<UserRoleAssignment[]>([])
+
+// TODO: Eliminate test data
+userRoleAssignments.value = [
+    {
+        "user_role_xref_id": 1,
+        "user": {
+            "user_type_code": "I",
+            "cognito_user_id": "foo-test",
+            "user_name": "Foo Tester"
+        },
+        "role": {
+            "role_name": "Reviewer",
+            "role_type_code": "C"
+        }
+    },
+    {
+        "user_role_xref_id": 2,
+        "user": {
+            "user_type_code": "B",
+            "cognito_user_id": "bar-test",
+            "user_name": "Bar Tester"
+        },
+        "role": {
+            "role_name": "Submitter",
+            "role_type_code": "A",
+            "client_number": {
+                "client_name": "acme forestry",
+                "forest_client_number": "00009876"
+            }
+        }
+    },
+    {
+        "user_role_xref_id": 3,
+        "user": {
+            "user_type_code": "B",
+            "cognito_user_id": "longer_user_id",
+            "user_name": "ThisIsAVery LongNameHere"
+        },
+        "role": {
+            "role_name": "Submitter",
+            "role_type_code": "A",
+            "client_number": {
+                "client_name": "Long name of forestry company",
+                "forest_client_number": "12345678"
+            }
+        }
+    }
+
+]
+
+userRoleAssignments.value.forEach(assignment => {
+  if (assignment.user.user_type_code == "B") {
+    assignment.user.user_type_desc = "BCeID"
+  } else if (assignment.user.user_type_code == "I") {
+    assignment.user.user_type_desc = "IDIR"
+  }
+})
+
+// TODO: Need API
+// userRoleAssignments.value = await apiService.getUserRoleAssignments(selectedApplication.value?.application_id)
+
+const userFilter = ref<string>()
+const roleFilter = ref<string>()
+const forestClientFilter = ref<string>()
+
+function showingMessage(): string {
+  let visible = 0;
+  userRoleAssignments.value.forEach(assignment => {
+    if (filterIncludes(assignment)) {
+      visible++
+    }
+  })
+
+  return "Showing " + visible + " of " + userRoleAssignments.value.length + " records"
+}
+
+function filterIncludes(userRoleAssignment: UserRoleAssignment):boolean {
+
+  // TODO: Review this logic.
+
+  if (userFilter.value != null) {
+    if (!userRoleAssignment.user.cognito_user_id.toLocaleUpperCase().includes(userFilter.value.toLocaleUpperCase())) {
```

Comment:
Hm... might not be correct using "cognito_user_id" for filtering.

Reply:
I agree, I've raised the question on Kevin's PR - but right now that's the only user id I'm getting back.

Reply:
Or maybe you mean filter by user name instead. Yes, that's one of the adjustments I should have considered after adding user name to the display.

Reply:
I am thinking of using "domain"+"user_name" (just like what you display), would that work better for now?

Reply:
I think we have differing defintiions of user name. My definition - using IDIR as an example:
user ID : bvandegr
user name: Basil Vandegriend

So we could combine domain + user ID (not user name). But if you want to sort the User ID column to find a particular ID, putting the domain first interferes with that.

Reply:
I see, can we clarify "User Name" "User Id" label for where the source columns are tomorrow from database point of view?
I don't see you are able to obtain "User Name" as what you indicated at the moment for MVP. The "User Name" is populated as "User ID" in database.